src/BuildingRegistry.Projections.Wfs/BuildingUnit/BuildingUnitProjections.cs

```diff
@@ -0,0 +1,467 @@
+                await AddBuildingUnit(context, message.Message.BuildingId, message.Message.BuildingUnitId, BuildingUnitFunction.Unknown, message.Message.Provenance.Timestamp, ct);
+            });
+
+            When<Envelope<BuildingUnitWasAddedToRetiredBuilding>>(async (context, message, ct) =>
+            {
+                await AddBuildingUnit(context, message.Message.BuildingId, message.Message.BuildingUnitId, BuildingUnitFunction.Unknown, message.Message.Provenance.Timestamp, ct);
+                var addedUnit = await context.BuildingUnits.FindAsync(message.Message.BuildingUnitId, cancellationToken: ct);
+                var building = await context.BuildingUnitsBuildings.FindAsync(message.Message.BuildingId, cancellationToken: ct);
+
+                if (building.BuildingRetiredStatus == BuildingStatus.NotRealized)
+                    addedUnit.Status = GebouweenheidStatus.NietGerealiseerd.ToString();
+                else
+                    addedUnit.Status = GebouweenheidStatus.Gehistoreerd.ToString();
+            });
+
+            When<Envelope<BuildingUnitWasReaddedByOtherUnitRemoval>>(async (context, message, ct) =>
+            {
+                await AddBuildingUnit(context, message.Message.BuildingId, message.Message.BuildingUnitId, BuildingUnitFunction.Unknown, message.Message.Provenance.Timestamp, ct);
+            });
+
+            When<Envelope<CommonBuildingUnitWasAdded>>(async (context, message, ct) =>
+            {
+                await AddBuildingUnit(context, message.Message.BuildingId, message.Message.BuildingUnitId, BuildingUnitFunction.Common, message.Message.Provenance.Timestamp, ct);
+            });
+
+            When<Envelope<BuildingUnitWasRemoved>>(async (context, message, ct) =>
+            {
+                var buildingUnit = await context.BuildingUnits.FindAsync(message.Message.BuildingUnitId, cancellationToken: ct);
+                context.BuildingUnits.Remove(buildingUnit);
+            });
+
+            When<Envelope<BuildingUnitBecameComplete>>(async (context, message, ct) =>
+            {
+                var buildingUnit = await context.BuildingUnits.FindAsync(message.Message.BuildingUnitId, cancellationToken: ct);
+                if (buildingUnit != null)
+                {
+                    buildingUnit.IsComplete = true;
+                    SetVersion(buildingUnit, message.Message.Provenance.Timestamp);
+                }
+            });
+
+            When<Envelope<BuildingUnitBecameIncomplete>>(async (context, message, ct) =>
+            {
+                var buildingUnit = await context.BuildingUnits.FindAsync(message.Message.BuildingUnitId, cancellationToken: ct);
+                if (buildingUnit != null)
+                {
+                    buildingUnit.IsComplete = false;
+                    SetVersion(buildingUnit, message.Message.Provenance.Timestamp);
+                }
+            });
+
+            #endregion
+
+            #region BuildingUnitStatus
```

Comment:
In Building you put them nicely in constants, can you do this here too and use them in all the status handlers below?